pytest_fixtures/core/sat_cap_factory.py

```diff
@@ -165,3 +165,11 @@ def parametrized_enrolled_sat(
     new_sat.unregister()
     new_sat.teardown()
     Broker(hosts=[new_sat]).checkin()
+
+
+@pytest.fixture(scope='module')
+def pre_upgrade_sat_host(module_satellite_factory):
```

Comment:
This sat would be used between pre and post as well, so I would recommend you to rename this to `module_upgrade_sat_host` or directly use `module_satelite_factory`.

Reply:
Also, this satellite should be of a older version, so that we upgrade to the latest and see if the things are good post upgrade.

Reply:
I think the `pre` prefix word has the advantage to know whether this test is related to scenario upgrade automation. We also have upgrade automation I just thought we should not mix that well. These are anyways n-1 version satellites understood by the name itself. could keep the name as `module_pre_upgrade_sat_host` what do you say?